builtin/logical/pki/cert_util.go

```diff
@@ -216,22 +218,12 @@ func fetchCertBySerial(req *logical.Request, prefix, serial string) (*logical.St
 	}
 
 	// No point checking these, no old/new style colons/hyphens
```

Comment:
I don't think this comment makes sense any more. We should keep the check with `path` for clarity, or update the comment to say why return if `legacyPath` is empty.